backend/graphql/types/testType.ts

```diff
@@ -16,43 +44,61 @@ const testType = gql`
     answerIndex: Float!
   }
 
-  input NumericQuestionMetadataInput {
+  input MultiSelectMetadataInput {
+    options: [String!]!
+    answerIndices: [Float!]!
+  }
+
+  type MultiSelectMetadata {
+    options: [String!]!
+    answerIndices: [Float!]!
+  }
+
+  input ShortAnswerMetadataInput {
     answer: Float!
   }
 
-  type NumericQuestionMetadata {
+  type ShortAnswerMetadata {
     answer: Float!
   }
 
-  union QuestionMetadata = MultipleChoiceMetadata | NumericQuestionMetadata
+  union QuestionComponentMetadata =
+      QuestionTextMetadata
+    | TextMetadata
+    | ImageMetadata
+    | MultipleChoiceMetadata
+    | MultiSelectMetadata
+    | ShortAnswerMetadata
 
-  type Question {
-    questionType: QuestionTypeEnum!
-    questionPrompt: String!
-    questionMetadata: QuestionMetadata!
+  type QuestionComponent {
+    type: QuestionComponentTypeEnum!
+    metadata: QuestionComponentMetadata!
   }
 
-  input QuestionInput {
-    questionType: QuestionTypeEnum!
-    questionPrompt: String!
-    questionMetadataMultipleChoice: MultipleChoiceMetadataInput
-    questionMetadataNumericQuestion: NumericQuestionMetadataInput
+  input QuestionComponentInput {
+    type: QuestionComponentTypeEnum!
+    questionTextMetadata: QuestionTextMetadataInput
+    textMetadata: TextMetadataInput
+    imageMetadata: ImageMetadataInput
+    multipleChoiceMetadata: MultipleChoiceMetadataInput
+    multiSelectMetadata: MultiSelectMetadataInput
+    shortAnswerMetadata: ShortAnswerMetadataInput
   }
```

Comment:
I know this is the pattern that we followed before, but is the type different for `ResponseDTO` and `RequestDTO`?